src/sedcli_main.c

```diff
@@ -761,37 +751,36 @@ static int handle_setpw(void)
 	return ret;
 }
 
-static int check_current_levl0_discv(void)
+static int check_current_levl0_discv(struct sed_device *dev)
 {
-	struct sed_opal_level0_discovery *discv = NULL;
 	int ret;
+	struct sed_opal_level0_discovery discv = { 0 };
 
-	discv = malloc(sizeof(*discv));
-	if (discv == NULL) {
-		sedcli_printf(LOG_ERR, "Memory not allocated.\n");
-		return -ENOMEM;
-	}
-
-	ret = sed_level0_discovery(discv);
+	ret = sed_level0_discovery(dev, &discv);
 	if (ret) {
-		if (ret == -EOPNOTSUPP)
-			sedcli_printf(LOG_ERR, "Command NOT supported for this "
-					"interface.\n");
-		else
-			sedcli_printf(LOG_ERR, "Error in level0 discovery\n");
-		goto free_mem;
+		if (ret == -EOPNOTSUPP) {
+			sedcli_printf(LOG_ERR, "Level0 discovery not supported "
```

Comment:
sounds like this is should be logged as LOG_WARNING vs LOG_ERR, because you allow operations

Reply:
will do it